yt_dlp/extractor/tenplay.py

```diff
@@ -113,3 +113,63 @@ def _real_extract(self, url):
             'uploader': 'Channel 10',
             'uploader_id': '2199827728001',
         }
+
+
+class TenPlaySeasonIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?10play\.com\.au/(?P<show>[^/?#]+)/episodes/(?P<season>[^/?#]+)/?(?:$|[?#])'
+
+    _TESTS = [{
+        'url': 'https://10play.com.au/masterchef/episodes/season-14',
+        'info_dict': {
+            'title': 'Season 14',
+            'id': 'MjMyOTIy',
+        },
+        'playlist_count': 64,
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        'url': 'https://10play.com.au/the-bold-and-the-beautiful-fast-tracked/episodes/season-2022',
+        'info_dict': {
+            'title': 'Season 2022',
+            'id': 'Mjc0OTIw',
+        },
+        'playlist_count': 256,
+        'params': {
+            'skip_download': True,
+        }
+    }]
+
+    def _entries(self, show, season, load_more_url):
+        skip_ids = []
+        has_more = True
+
+        base_url = f'https://10play.com.au/{show}/episodes/{season}'
+
+        while has_more:
+            episodes_carousel = self._download_json(
+                load_more_url, f'{show}/{season}', query={'skipIds[]': skip_ids},
+                note=f'Fetching episodes {len(skip_ids)}+')
+
+            has_more = episodes_carousel['hasMore']
+            episodes_chunk = episodes_carousel['items']
+            skip_ids += [ep['id'] for ep in episodes_chunk]
+
+            yield from [urljoin(base_url, ep['cardLink']) for ep in episodes_chunk]
```

Comment:
- we can delegate the `urljoin`-ing to the `playlist_from_matches` getter
- displaying episode count during API calls may be misleading since we don't know which order the episodes are in; counting pages with itertools seems more straightforward
```suggestion
    def _entries(self, load_more_url, display_id=None):
        skip_ids = []
        for page in itertools.count(1):
            episodes_carousel = self._download_json(
                load_more_url, display_id, query={'skipIds[]': skip_ids},
                note=f'Fetching episodes page {page}')

            episodes_chunk = episodes_carousel['items']
            skip_ids.extend(ep['id'] for ep in episodes_chunk)

            for ep in episodes_chunk:
                yield ep['cardLink']
            if not episodes_carousel['hasMore']:
                break
```
import `iterools`